src/main.cpp

```diff
@@ -5605,7 +5602,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         CTransaction tx;
         vRecv >> tx;
 
-        std::shared_ptr<CTxVerificationQueueEntry> ctvqe = createCTxVerificationQueueEntry(tx, pfrom->GetId());
+        CTxVerificationQueueEntry* ctvqe = createCTxVerificationQueueEntry(tx, pfrom->GetId());
```

Comment:
There is no need to have ptrs allocated on the heap in the queue, but objects instead, for example:

        CTxVerificationQueueEntry ctxvqe;
        ctxvqe.setTX(tx);
        ctxvqe.setNodeId(pfrom->GetId());

Reply:
please checkout this commit

https://github.com/andreysobol/zen/commit/13eac27e36ebabd9a275328c5f7df03d858b3976

I remove pointer. Can you please doublecheck scenario of "segmentation fault"?